packages/bull-vault/src/BullStrategy.sol

```diff
@@ -137,6 +173,7 @@ contract BullStrategy is ERC20, LeverageBull {
      * @param _bullAmount amount of bull token to redeem
      */
     function withdraw(uint256 _bullAmount) external {
+        require(!IController(powerTokenController).isShutDown(), "BS7");
```

Comment:
Can't we just check with `hasRedeemedInShutdown` ?

Reply:
hasRedeemedInShutdown is only when its 100% redeemed, we also need to prevent withdraw() when its 50% redeemed